src/raster/processing/dem/options.rs

```diff
@@ -0,0 +1,172 @@
+    Horn,
+    ZevenbergenThorne,
+}
+
+impl Display for DemSlopeAlg {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        f.write_fmt(format_args!("{self:?}"))
+    }
+}
+
+macro_rules! common_dem_options {
+    () => {
+        /// Specify which band in the input [`Dataset`][crate::Dataset] to read from.
+        ///
+        /// Defaults to the first band.
+        pub fn with_input_band(&mut self, band: NonZeroUsize) -> &mut Self {
+            self.input_band = Some(band);
+            self
+        }
+
+        /// Fetch the specified input band to read from.
+        pub fn input_band(&self) -> Option<NonZeroUsize> {
+            self.input_band
+        }
+
+        /// Explicitly specify output raster format.
+        ///
+        /// This is equivalent to the `-of <format>` CLI flag accepted by many GDAL tools.
+        ///
+        /// The value of `format` must be the identifier of a driver supported by the runtime
+        /// environment's GDAL library (e.g. `COG`, `JPEG`, `VRT`, etc.). A list of these identifiers
+        /// is available from `gdalinfo --formats`:
+        ///
+        /// ```text
+        /// ❯ gdalinfo --formats
+        /// Supported Formats:
+        ///   VRT -raster,multidimensional raster- (rw+v): Virtual Raster
+        ///   DERIVED -raster- (ro): Derived datasets using VRT pixel functions
+        ///   GTiff -raster- (rw+vs): GeoTIFF
+        ///   COG -raster- (wv): Cloud optimized GeoTIFF generator
+        ///   NITF -raster- (rw+vs): National Imagery Transmission Format
+        /// ...
+        /// ```
+        ///
+        pub fn with_output_format(&mut self, format: &str) -> &mut Self {
+            self.output_format = Some(format.to_owned());
+            self
+        }
+
+        /// Fetch the specified output format driver identifier.
+        pub fn output_format(&self) -> Option<String> {
+            self.output_format.clone()
+        }
+
+        /// Compute values at image edges.
+        ///
+        /// If true, causes interpolation of values at image edges or if a no-data value is found
+        /// in the 3x3 processing window.
+        pub fn with_compute_edges(&mut self, state: bool) -> &mut Self {
+            self.compute_edges = state;
+            self
+        }
+
+        /// Fetch the compute edges mode.
+        pub fn compute_edges(&self) -> bool {
+            self.compute_edges
+        }
+
+        /// Additional generic options to be included.
+        pub fn with_additional_options(&mut self, extra_options: CslStringList) -> &mut Self {
+            self.additional_options.extend(&extra_options);
+            self
+        }
+
+        /// Fetch additional options.
+        pub fn additional_options(&self) -> &CslStringList {
+            &self.additional_options
+        }
+
+        /// Private utility to convert common options into [`CslStringList`] options.
+        fn store_common_options_to(&self, opts: &mut CslStringList) {
```

Comment:
We can return a `Result<()>` from this one and `expect` on the caller side instead of doing 5 `unwrap`s.

Reply:
https://github.com/metasim/gdal/pull/1

Reply:
I thought of this more like using `unsafe`: As a human with a static set of invariants, I'm guaranteeing those calls will never fail.

Reply:
Yeah, same thing as `dataset.rasterband(0)`. But I think it looks cleaner like this, and it will never matter in practice.